File: test/kotlin/org/aguinore/crackInterview/CrackInterviewTest.kt
package org.aguinore.crackInterview

import org.junit.jupiter.api.Test
import org.junit.jupiter.api.TestInstance
import kotlin.test.assertEquals

@TestInstance(TestInstance.Lifecycle.PER_METHOD)
class CrackInterviewKotlinTest {
    @Test
    fun numberOfPairsTest() {
        val arr = intArrayOf(10, 20, 20, 10, 10, 30, 50, 10, 20)
        val result = 3
        assertEquals(result, CrackInterviewKotlin.numberOfPairs(arr))
    }

    @Test
    fun numberOfPairsTest2() {
        val arr = intArrayOf(1, 2, 1, 2, 1, 3, 2)
        val result = 2
        assertEquals(result, CrackInterviewKotlin.numberOfPairs(arr))
    }

    @Test
    fun minimumDistanceBetweenPairs() {
        val arr = intArrayOf(7, 1, 3, 4, 1, 7)
        val result = 3
        assertEquals(result, CrackInterviewKotlin.minimumDistances(arr))
    }

    @Test
    fun minimumDistanceBetweenPairs2() {
        val arr = intArrayOf(1, 2, 3, 4, 10)
        val result = -1
        assertEquals(result, CrackInterviewKotlin.minimumDistances(arr))
    }
}

File: test/python/org/aguinore/crackInterview/crackInterviewTest.py
import unittest

from src.main.python.org.aguinore.crackInterview import crack_interview


class TestCrackInterveiw(unittest.TestCase):

    def test_number_of_pairs(self):
        arr = [10, 20, 20, 10, 10, 30, 50, 10, 20]
        result = 3
        actual = crack_interview.number_of_pairs(arr)
        self.assertEqual(actual, result, "expected: " + str(result) + " actual: " + str(actual))

    def test_number_of_pairs2(self):
        arr = [1, 2, 1, 2, 1, 3, 2]
        result = 2
        self.assertEqual(crack_interview.number_of_pairs(arr), result)